platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/engine/EcdsaSecp256k1Verifier.java

```diff
@@ -152,27 +166,29 @@ public static byte[] minifiedPositiveBigEndian(final byte[] v) {
      * @return true if the signature is valid
      */
     public boolean verify(final byte[] rawSig, final byte[] msg, final byte[] pubKey) {
-        try {
-            algorithm.initVerify(asCryptographic(pubKey));
-            algorithm.update(msg);
-            final byte[] asn1DerSig = asn1DerEncode(rawSig);
-            return algorithm.verify(asn1DerSig);
-        } catch (InvalidKeySpecException | SignatureException | InvalidKeyException e) {
+        final ThreadLocalCache cache = CACHE.get();
+        // convert signature to native format
+        final LibSecp256k1.secp256k1_ecdsa_signature nativeSignature = cache.signature;
+        final var signatureParseResult =
+                LibSecp256k1.secp256k1_ecdsa_signature_parse_compact(LibSecp256k1.CONTEXT, nativeSignature, rawSig);
+        if (signatureParseResult != 1) {
             logger.debug(
-                    TESTING_EXCEPTIONS.getMarker(),
-                    () -> "Failure while verifying signature [ publicKey = %s, rawSig = %s ]"
-                            .formatted(hex(pubKey), hex(rawSig)),
-                    e);
+                    TESTING_EXCEPTIONS.getMarker(), () -> "Failed to parse signature [ publicKey = %s, rawSig = %s ]"
+                            .formatted(hex(pubKey), hex(rawSig)));
             return false;
         }
-    }
-
-    private ECPublicKey asCryptographic(final byte[] pubKey) throws InvalidKeySpecException {
-        final BigInteger xCoord = new BigInteger(1, Arrays.copyOfRange(pubKey, 0, EC_COORD_SIZE));
-        final BigInteger yCoord = new BigInteger(1, Arrays.copyOfRange(pubKey, EC_COORD_SIZE, pubKey.length));
-
-        final ECPoint curvePoint = new ECPoint(xCoord, yCoord);
-        final ECPublicKeySpec ecPublicKeySpec = new ECPublicKeySpec(curvePoint, secp256k1Params);
-        return (ECPublicKey) ecKeyFactory.generatePublic(ecPublicKeySpec);
+        // Normalize the signature to lower-S form. This will return 1 if the signature was normalized, 0 otherwise.
+        LibSecp256k1.secp256k1_ecdsa_signature_normalize(LibSecp256k1.CONTEXT, nativeSignature, nativeSignature);
+        // convert public key to input format
+        final byte[] publicKeyInput = cache.uncompressedPublicKeyInput;
+        System.arraycopy(pubKey, 0, publicKeyInput, 1, ECDSA_UNCOMPRESSED_KEY_SIZE);
+        // convert public key to native format
+        final LibSecp256k1.secp256k1_pubkey pubkey = cache.publicKey;
+        final int keyParseResult = LibSecp256k1.secp256k1_ec_pubkey_parse(
+                LibSecp256k1.CONTEXT, pubkey, publicKeyInput, publicKeyInput.length);
+        if (keyParseResult != 1) throw new RuntimeException("Failed to parse public key");
```

Comment:
Why not just return `false` if the key can't be parsed or is invalid? Semantically, the signature can not be verified with the provided key. A `RuntimeException` is expensive and may be too disruptive here.

Reply:
Thanks fixed